ui/app/components/wizard/features-selection.js

```diff
@@ -6,18 +6,33 @@ import { FEATURE_MACHINE_TIME } from 'vault/helpers/wizard-constants';
 export default Component.extend({
   wizard: service(),
   version: service(),
+  permissions: service(),
 
   init() {
     this._super(...arguments);
     this.maybeHideFeatures();
   },
 
   maybeHideFeatures() {
+    let features = this.get('allFeatures');
+    features.forEach(feat => {
+      feat.disabled = this.doesNotHavePermission(feat.requiredPermissions);
+    });
+
     if (this.get('showReplication') === false) {
       let feature = this.get('allFeatures').findBy('key', 'replication');
       feature.show = false;
     }
   },
+
+  doesNotHavePermission(requiredPermissions) {
+    return !Object.keys(requiredPermissions).every(path => {
+      return requiredPermissions[path].every(capability => {
```

Comment:
Is this second loop necessary? Could we just do: 
`return this.permissions.hasPermission(path, requiredPermission[path])`
since hasPermission is going to `every` over that array?

Reply:
yes! good catch. i forgot that `hasPermissions` accepts an array. fixing now.